DOCGroup/autobuild
# Update JUnit XML Reports

Added the hostname field to the \<testsuite\> element.
Added \<properties\> and child \<property\> elements to the \<testsuite\> element, containing “non-standard” data elements:
  - commits: GitHub commit hashes.
  - log_file: URL or relative path to the full log file.

Two ways to test PR#44:
1. Use scoreboard.pl:
====================
1.1 Create and cd to a test directory that contains the build_logs directory.  The build_logs directory contains a test.xml file and at least one build directory:
$ ls -R build_logs
build1  test.xml
build_logs/build1: 2021_03_07_16_45.txt  2021_03_08_16_59.txt  2021_03_09_17_33.txt  latest.txt  status.txt  post
$ cat build_logs/test.xml
\<scoreboard\>
  \<group\>
    \<name\>Group 1\</name\>
    \<build\>
      \<name\>build1\</name\>
    \</build\>
  \</group\>
\</scoreboard\>
1.2 Run scoreboard.pl:
$ \<path to\>/autobuild/scoreboard.pl -c -v -u -d build_logs -o test.html -f build_logs/test.xml
1.3 Verify that the *_JUnit.xml files contain the property entries of git commit and of path to the full log html file.

2. Use autobuild.pl:
====================
2.1 In the OpenDDS root, create a workspace directory that contains a log file and a config file:
$ ls -R workspace/
2021_03_09_17_33.txt    config_process_logs_only.xml
Replace 'OpenDDS_ROOT' with the actual path to the OpenDDS root:
$ cat workspace/config_process_logs_only.xml
\<autobuild\>
\<configuration\>
\<variable name="project_root" value="OpenDDS_ROOT"/\>
\<variable name="root" value="OpenDDS_ROOT/workspace"/\>
\<variable name="log_root" value="OpenDDS_ROOT/workspace"/\>
\<variable name="log_file" value="2021_03_09_17_33.txt"/\>
\<variable name="junit_xml_output" value="2021_03_09_17_33"/\>
\</configuration\>
\<command name="process_logs" options="prettify"/\>
\</autobuild>
2.2 Run autobuild.pl:
$ \<path to\>/autobuild/autobuild.pl workspace/config_process_logs_only.xml
2.3 Verify that the *_JUnit.xml file contains the property entries of git commit and of path to the full log html file. 
$ cat workspace/2021_03_09_17_33_JUnit.xml | grep "property"
      \<property name="GIT_COMMIT_OPENDDS" value="6c45822ce73ac6a1e9370e48ed7ac836a91b6ffd"/\>
      \<property name="log_file" value="OpenDDS_ROOT/workspace/2021_03_09_17_33_Full.html"/\>
